guides/source/asset_pipeline.md

```diff
@@ -100,194 +76,289 @@ content into the name, usually at the end. For example a CSS file `global.css`
-are not special - any path under `assets/*` will be searched.
+As you can see, the two packages react and react-dom resolve to a total of four dependencies, when resolved via the jspm default.
 
-For example, these files:
+Now you can use these in your `application.js` entrypoint like you would any other module:
 
+```javascript
+import React from "react"
+import ReactDOM from "react-dom"
 ```
-app/assets/javascripts/home.js
-lib/assets/javascripts/moovinator.js
-vendor/assets/javascripts/slider.js
-vendor/assets/somepackage/phonebox.js
+
+You can also designate a specific version to pin:
+
+```sh
+./bin/importmap pin react@17.0.1
+Pinning "react" to https://ga.jspm.io/npm:react@17.0.1/index.js
+Pinning "object-assign" to https://ga.jspm.io/npm:object-assign@4.1.1/index.js
 ```
 
-would be referenced in a manifest like this:
+Or even remove pins:
 
-```js
-//= require home
-//= require moovinator
-//= require slider
-//= require phonebox
+```sh
+./bin/importmap unpin react
+Unpinning "react"
+Unpinning "object-assign"
 ```
 
-Assets inside subdirectories can also be accessed.
+You can control the environment of the package for packages with separate "production" (the default) and "development" builds:
 
+```sh
+./bin/importmap pin react --env development
+Pinning "react" to https://ga.jspm.io/npm:react@17.0.2/dev.index.js
+Pinning "object-assign" to https://ga.jspm.io/npm:object-assign@4.1.1/index.js
 ```
-app/assets/javascripts/sub/something.js
+
+You can also pick an alternative, supported CDN provider when pinning, like `unpkg` or `jsdelivr` (`jspm` is the default):
+
+```sh
+./bin/importmap pin react --from jsdelivr
+Pinning "react" to https://cdn.jsdelivr.net/npm/react@17.0.2/index.js
 ```
 
-is referenced as:
+Remember, though, that if you switch a pin from one provider to another, you may have to clean up dependencies added by the first provider that isn't used by the second provider.
 
-```js
-//= require sub/something
+Run `./bin/importmap` to see all options.
+
+Note that this command is merely a convenience wrapper to resolving logical package names to CDN URLs. You can also just lookup the CDN URLs yourself, and then pin those. For example, if you wanted to use Skypack for React, you could just add the following to `config/importmap.rb`:
+
+```ruby
+pin "react", to: "https://cdn.skypack.dev/react"
 ```
 
-You can view the search path by inspecting
-[`Rails.application.config.assets.paths`][`config.assets.paths`] in the Rails console.
+### Preloading pinned modules
 
-Besides the standard `assets/*` paths, additional (fully qualified) paths can be
-added to the pipeline in `config/initializers/assets.rb`. For example:
+To avoid the waterfall effect where the browser has to load one file after another before it can get to the deepest nested import, importmap-rails supports [modulepreload links](https://developers.google.com/web/updates/2017/12/modulepreload). Pinned modules can be preloaded by appending `preload: true` to the pin.
```

Comment:
Is there some generic advice we can give on which modules you should/shouldn't preload?

Reply:
I did some Googling and I didn't find anything super useful, most clarifying for me was this answer:
https://stackoverflow.com/a/63924789/9984998

And I am thinking of adding such a sentence to the guide:
> Preloading is extremely useful in case you know that you'd need a specific part of the code, but you don't want to execute it immediately.

Getting to specific about the library might be difficult, what do you think?

Reply:
How about this:

> It's a good idea to preload libraries or frameworks that are used throughout your app, as this will tell the browser to download them sooner.

Stimulus and Turbo get preloaded by default

https://github.com/hotwired/stimulus-rails/blob/498902a3687d32bd6589f7d3c7e7d72eabd822c0/lib/install/stimulus_with_importmap.rb#L15

And I think this explanation is okay for why.